rfcs/convergence/contract-for-trigger-pattern.md

```diff
@@ -0,0 +1,134 @@
+# RFC: Creating a structured contract for Trigger - like patterns
+
+@ling1726
+
+## Problem statement
+
+```tsx
+<Tooltip>
+  <Button />
+</Tooltip>
+
+<Menu>
+  <MenuTrigger>
+    <Button />
+  </MenuTrigger>
+</Menu>
+
+<Popover>
+  <PopoverTrigger>
+    <Button />
+  </PopoverTrigger>
+</Popover>
+```
+
+The above examples show the adoption of a `Trigger` pattern that was inspired from [react-spectrum](https://react-spectrum.adobe.com/react-spectrum/MenuTrigger.html), where the library component will clone the child and add appropriate event handlers/props to handle complex interactions to simplify the lives of our users. These can be:
```

Comment:
_nit:_ I don't think that the first part of statement is correct, as the pattern with usage of `.cloneElement()` for this purpose is widely adopted:
- Reach UI: https://reach.tech/tooltip
- Semantic UI React: https://react.semantic-ui.com/modules/popup/
- Material UI: https://material-ui.com/components/click-away-listener/
- Fluent UI Northstar: https://fluentsite.z22.web.core.windows.net/0.56.0/components/tooltip/definition